sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/Source.scala

```diff
@@ -29,8 +30,13 @@ trait Source  {
   /** Returns the schema of the data from this source */
   def schema: StructType
 
+  /** Returns the maximum available offset for this source. */
+  def getOffset: Option[Offset]
+
   /**
-   * Returns the next batch of data that is available after `start`, if any is available.
+   * Returns the data that is is between the offsets (`start`, `end`].  When `start` is `None` then
+   * the batch should begin with the first available record.  This method must always return the
```

Comment:
Shouldnt say first available record. In this case, the source is free to choose the point from which it wants to return the data, first available, or latest available (e.g. Kinesis can choose to return from TRIM_HORIZON, or LATEST)